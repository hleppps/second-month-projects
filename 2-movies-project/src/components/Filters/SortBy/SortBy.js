import React from "react";
import { Form } from "react-bootstrap";

const SortBy = React.memo((props) => {
  const onChangeHandler = (event) => {
    props.setSortingFilter(event.target.value);
  };

  return (
    <Form.Group className="my-3">
      <Form.Label>Sort by:</Form.Label>
      <Form.Select
        data-testid="sort-select"
        aria-label="Default select example"
        onChange={onChangeHandler}
        value={props.filteredSorting}
      >
        <option value="popularity.desc">Most Popular</option>
        <option value="popularity.asc">Least Popular</option>
        <option value="vote_average.desc">Most Wanted</option>
        <option value="vote_average.asc">Least Wanted</option>
      </Form.Select>
    </Form.Group>
  );
});

export default SortBy;
