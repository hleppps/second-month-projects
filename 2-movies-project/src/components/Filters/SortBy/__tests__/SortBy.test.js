import "@testing-library/jest-dom";
import { cleanup, fireEvent, render } from "@testing-library/react";
import SortBy from "../SortBy";

afterEach(cleanup);

describe("SortBy component", () => {
  it("renders year value from props", () => {
    const sortingType = "vote_average.asc";

    const { getByTestId } = render(<SortBy filteredSorting={sortingType} />);

    expect(getByTestId("sort-select")).toHaveValue(sortingType);
  });

  it("calls the year change function on change", () => {
    const sortingChangeHandler = jest.fn();

    const { getByTestId } = render(
      <SortBy setSortingFilter={sortingChangeHandler} />
    );

    fireEvent.change(getByTestId("sort-select"));

    expect(sortingChangeHandler).toBeCalled();
  });

  it("matches a snapshot", () => {
    const { asFragment } = render(<SortBy />);
    expect(asFragment(<SortBy />)).toMatchSnapshot();
  });
});
