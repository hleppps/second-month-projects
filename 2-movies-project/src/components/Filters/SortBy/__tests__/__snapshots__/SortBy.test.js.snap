// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy component matches a snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-3"
  >
    <label
      class="form-label"
    >
      Sort by:
    </label>
    <select
      aria-label="Default select example"
      class="form-select"
      data-testid="sort-select"
    >
      <option
        value="popularity.desc"
      >
        Most Popular
      </option>
      <option
        value="popularity.asc"
      >
        Least Popular
      </option>
      <option
        value="vote_average.desc"
      >
        Most Wanted
      </option>
      <option
        value="vote_average.asc"
      >
        Least Wanted
      </option>
    </select>
  </div>
</DocumentFragment>
`;
